src/frontend/src/app/HomePageContext.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+
+interface HomePageContextProps {
+  onPNMHomePage: boolean;
+  setOnPNMHomePage: (value: boolean) => void;
+  onGuestHomePage: boolean;
+  setOnGuestHomePage: (value: boolean) => void;
+}
+
+const HomePageContext = createContext<HomePageContextProps | undefined>(undefined);
+
+export const HomePageProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const [onGuestHomePage, setOnGuestHomePage] = useState<boolean>(() =>
+    JSON.parse(localStorage.getItem('onGuestHomePage') || 'true')
```

Comment:
yeah i feel like if were doing this we might as well just put it in the url instead, make our lives significantly easier